### PR TITLE
Added Architecture Overview

### PR DESCRIPTION
**Pull request by othylmann**
Greetings all. I created a first architecture overview. I think the AWS is pretty ok, but onprem might be a bit lagging still. The files for the diagrams are in Google Drive under Product Architecture.

### CODE REVIEW

**Comment by puja108** [comment]
I'll be doing a slight revision on this, correcting some wording and working on clarity today.

@teemow do you have time for a quick sanity check? Or have you seen this already?

**Comment by puja108** [comment]
finished my revision for now, maybe @calvix or @r7vme can fact check a bit, not urgent.

File: content/reference/giantswarm-onprem-architecture/index.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+## Giant Swarm On Premise Host Cluster
+
+![On Premise Host Cluster Architecture](/img/architecture-onprem-host-cluster.png)
+
+Think of the Host Cluster as the Giant Swarm team’s Kubernetes Cluster. The worker nodes are where your Guest Clusters will end up on, controlled by our [kvm-operator](https://github.com/giantswarm/kvm-operator/) running inside the Host Cluster. The above diagram shows our PoC setup in that keeps the Host Cluster (including monitoring) on 3 machines. Based on amount and sizes of Guest Clusters, we recommend having distinct monitoring machines once we go into production. We further recommend having at least 5 worker machines. Any additional machines will also be set up as worker nodes and made available for Guest Clusters to be used by end users. 
+
+We access all machines, as well as the Kubernetes API of the Host Cluster, via the Bastion host. We connect your load balancer for access to the Giant Swarm API, Happa, and our Monitoring, optionally adding your Identity Management System for authentication. Traffic towards all Guest Clusters is routed via the Host Cluster first, as it knows where exactly all the Guest Clusters are at any point in time.
```
Comment (teemow) [inline content/reference/giantswarm-onprem-architecture/index.md]: We didn't atlk about bastions before in the on premise setup. They are basically optional. So we connect through VPN and/or bastion.

**Comment by teemow** [comment]
Great! LGTM

**Comment by othylmann** [comment]
Thank you for pushing this through in my absence :)

**Pull request merged**